From be1c610b0b9d4598fcda35431d0cd3e70dc2e3bc Mon Sep 17 00:00:00 2001
From: Adheer Chandravanshi <adheer.chandravanshi@qlogic.com>
Date: Fri, 18 Sep 2015 05:15:50 -0400
Subject: [PATCH] iscsiuio: Add QLogic Vendor ID to support newer NX2 HBAs

The newer 57840 NX2 HBAs have started using QLogic Vendor ID.
Adding the support for it.

Signed-off-by: Adheer Chandravanshi <adheer.chandravanshi@qlogic.com>
---
 iscsiuio/src/unix/libs/bnx2x.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/iscsiuio/src/unix/libs/bnx2x.c b/iscsiuio/src/unix/libs/bnx2x.c
index a364d76..c9e3436 100644
--- a/iscsiuio/src/unix/libs/bnx2x.c
+++ b/iscsiuio/src/unix/libs/bnx2x.c
@@ -120,6 +120,7 @@ static const char brcm_57840_2_20[] = "QLogic NetXtreme II BCM57840 2x"
  * PCI ID constants
  ******************************************************************************/
 #define PCI_VENDOR_ID_BROADCOM			0x14e4
+#define PCI_VENDOR_ID_QLOGIC			0x1077
 #define PCI_DEVICE_ID_NX2_57710			0x164e
 #define PCI_DEVICE_ID_NX2_57711			0x164f
 #define PCI_DEVICE_ID_NX2_57711E		0x1650
@@ -194,6 +195,14 @@ static const struct pci_device_id bnx2x_pci_tbl[] = {
 	 PCI_ANY_ID, PCI_ANY_ID, brcm_57840_MF},
 	{PCI_VENDOR_ID_BROADCOM, PCI_DEVICE_ID_NX2_57840_VF,
 	 PCI_ANY_ID, PCI_ANY_ID, brcm_57840_VF},
+	{PCI_VENDOR_ID_QLOGIC, PCI_DEVICE_ID_NX2_57840_4_10,
+	 PCI_ANY_ID, PCI_ANY_ID, brcm_57840_4_10},
+	{PCI_VENDOR_ID_QLOGIC, PCI_DEVICE_ID_NX2_57840_2_20,
+	 PCI_ANY_ID, PCI_ANY_ID, brcm_57840_2_20},
+	{PCI_VENDOR_ID_QLOGIC, PCI_DEVICE_ID_NX2_57840_MF,
+	 PCI_ANY_ID, PCI_ANY_ID, brcm_57840_MF},
+	{PCI_VENDOR_ID_QLOGIC, PCI_DEVICE_ID_NX2_57840_VF,
+	 PCI_ANY_ID, PCI_ANY_ID, brcm_57840_VF},
 };
 
 static struct iro e1_iro[2] = {
-- 
2.7.4

